google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineEnvironment.java

```diff
@@ -23,7 +23,8 @@
  */
 public enum AppEngineEnvironment {
   APP_ENGINE_STANDARD("appengine.environment.name.standard"),
-  APP_ENGINE_FLEX("appengine.environment.name.flexible");
+  APP_ENGINE_FLEX("appengine.environment.name.flexible"),
+  APP_ENGINE_FLEX_COMPAT("appengine.environment.name.flexible");
```

Comment:
I think the discussion/sematics problem shows up (and has shown up a lot already) any time we discuss any logic around flex compat in our plugin. I feel like we've always had to define first what we mean by the term "environment" before a meaningful discussion can be had.

I don't follow your point about the issue with renaming the class. Maybe AppEngineProjectType isn't the best alternative, but if the enum is not representing the _cloud_ environment, then AppEngineEnvironment is not the right name.